pkg/cache/application.go

```diff
@@ -395,7 +395,11 @@ func (app *Application) scheduleTasks(taskScheduleCondition func(t *Task) bool)
 	for _, task := range app.GetNewTasks() {
 		if taskScheduleCondition(task) {
 			// for each new task, we do a sanity check before moving the state to Pending_Schedule
-			if err := task.sanityCheckBeforeScheduling(); err == nil {
+			// if the task is not ready for scheduling, we keep it in New state
+			// if the task pod is bounded and have conflicting metadata, we move the task to Rejected state
+			err, rejectTask := task.sanityCheckBeforeScheduling()
```

Comment:
Perform a sanity check before move this task to Pending state.

Before this PR, sanity check only check PVC's readiness 
- If sanity check passed, move task state from 'New' -> 'Pending'
- If sanity check failed, task state remains in 'New' (Will be checked again in next schedule cycle)

After this PR (Sanity check check PVC and Pod Metadata)
- if sanity check passed, 'New' -> 'Pending'  
- if sanity check fails due to PVC -> 'New' (No change)
- if sanity check fails due to a unbound pod with inconsistent metadata (AppID/Label), move task state from 'New' to 'Rejected'

**Design decision**: Only reject unbound pods because we don't want to failed existing running pod after restart YK.